### PR TITLE
board: npcx: add npcx9m6f_evb board.

### PR DESCRIPTION
**Pull request by MulinChao**
This PR is the last one for porting npcx9 series driver to Zephyr. It still has the dependency with [PR34795](https://github.com/zephyrproject-rtos/zephyr/pull/34795) and [PR35287](https://github.com/zephyrproject-rtos/zephyr/pull/35287). Once they were landed, I will do rebasing to pass the sanity checks. 

So far, the driver implementation has passed the following driver/kernel test suites on npcx9m6f_evb board.
pwm
gpio
adc 
watchdog
i2c
uart
kernel/timer

Signed-off-by: Mulin Chao <mlchao@nuvoton.com>

### CODE REVIEW

File: boards/arm/npcx9m6f_evb/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
```
Comment (keith-zephyr) [inline boards/arm/npcx9m6f_evb/doc/index.rst:46]: The board DTS file also enables the following components:
PSL, PWM, ADC, eSPI, I2C, TACH
Plus power management is also supported.

File: boards/arm/npcx9m6f_evb/npcx9m6f_evb.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+identifier: npcx9m6f_evb
+name: Nuvoton NPCX9M6F EVB
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+ram: 64
+flash: 256
+supported:
```
Comment (keith-zephyr) [inline boards/arm/npcx9m6f_evb/npcx9m6f_evb.yaml:16]: This list should match the doc file.

Comment (MulinChao) [reply boards/arm/npcx9m6f_evb/npcx9m6f_evb.yaml:16]: Done.

File: boards/arm/npcx9m6f_evb/doc/index.rst (hunk 1 above)
Comment (MulinChao) [reply boards/arm/npcx9m6f_evb/doc/index.rst]: Done.

File: boards/arm/npcx9m6f_evb/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+UART1 is configured for serial logs.
+
+
+Programming and Debugging
+*************************
+
+This board comes with a Cortex ETM port which facilitates tracing and debugging
+using a single physical connection. In addition, it comes with sockets for
+JTAG only sessions.
```
Comment (sjg20) [inline boards/arm/npcx9m6f_evb/doc/index.rst:92]: JTAG-only

**Review by sjg20 — CHANGES_REQUESTED** [review]
in commit message:

"This CL introduces " can be "Introduce the "

**Review by keith-zephyr — APPROVED** [review]

File: boards/arm/npcx9m6f_evb/doc/index.rst (hunk 3 above)
Comment (MulinChao) [reply boards/arm/npcx9m6f_evb/doc/index.rst]: Done.

**Comment by MulinChao** [comment]
The failure of" Compliance" is caused by the dependency of [PR34795](https://github.com/zephyrproject-rtos/zephyr/pull/34795) and [PR35287](https://github.com/zephyrproject-rtos/zephyr/pull/35287). Once they're landed, I will do rebasing to fix it.

**Review by sjg20 — APPROVED** [review]

**Review by yperess — APPROVED** [review]

**Comment by MulinChao** [comment]
The failures of sanity checks are caused by the dependency of #35287. Once it was landed, I will do rebasing to fix it.

**Pull request merged**